File: 0404/solution.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def traverse(node, isLeft):
            if not node:
                return 0
            if not node.left and not node.right and isLeft:
                return node.val
            left = traverse(node.left, True)
            right = traverse(node.right, False)
            return left + right

        return traverse(root, False)

File: 0404/solution.ts
class TreeNode {
  val: number;
  left: TreeNode | null;
  right: TreeNode | null;
  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }
}

function sumOfLeftLeaves(node: TreeNode | null, isLeft = false): number {
  if (!node) {
    return 0;
  }

  if (!node.left && !node.right && isLeft) {
    return node.val;
  }

  return sumOfLeftLeaves(node.left, true) + sumOfLeftLeaves(node.right, false);
}
